src/BizHawk.Emulation.Cores/Computers/Doom/DSDAControllers.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System.Linq;
+
+using BizHawk.Common;
+using BizHawk.Emulation.Common;
+
+namespace BizHawk.Emulation.Cores.Computers.Doom
+{
+	public enum DoomControllerTypes
+	{
+		Doom,
+		Heretic,
+		Hexen
+	}
+
+	public interface IPort
+	{
+		byte Read(IController c);
+
+		int Read_Pot(IController c, int pot);
+
+		ControllerDefinition Definition { get; }
+
+		int PortNum { get; }
+	}
+
+	public class DoomController : IPort
+	{
+		public DoomController(int portNum)
+		{
+			PortNum = portNum;
+			Definition = new ControllerDefinition("Doom Input Format")
+			{
+				BoolButtons = BaseDefinition
+				.Select(b => $"P{PortNum} " + b)
+				.ToList()
+			}.AddAxis($"P{PortNum} Run Speed", (-50).RangeTo(50), 0)
+				.AddAxis($"P{PortNum} Strafing Speed", (-127).RangeTo(127), 0)
+				.AddAxis($"P{PortNum} Turning Speed", (-128).RangeTo(128), 0)
+				.AddAxis($"P{PortNum} Weapon Select", (0).RangeTo(7), 0)
+				.AddAxis($"P{PortNum} Mouse Running", (-128).RangeTo(128), 0)
+				.AddAxis($"P{PortNum} Mouse Turning", (-128).RangeTo(128), 0)

```

Comment:
Assuming these are stored as a byte, shouldn't all of these be `[-128, 127]`? (same for the other controllers below)

Reply:
fixed